homeassistant/helpers/entity_component.py

```diff
@@ -131,13 +131,16 @@ def _async_setup_platform(self, platform_type, platform_config,
         scan_interval = (platform_config.get(CONF_SCAN_INTERVAL) or
                          getattr(platform, 'SCAN_INTERVAL', None) or
                          self.scan_interval)
+        scan_sequential = (platform_config.get(CONF_SCAN_SEQUENTIAL) or
```

Comment:
We should not be able to configure this. The developer of the platform should decide this.

Reply:
On stuff like rest platform? So you can trottle the updates?

Reply:
I'm also somewhat inclined to say that in a perfect world we wouldn't worry about this at all in hass core. If an IOT device/gateway chokes on parallel requests, the third-party lib would ideally queue / throttle these requests appropriately. Sequential requests is really just an imperfect workaround. It might be still too much load for some devices, and some devices might handle up to 4 parallel requests just fine. We'd ideally just let the third-party libs manage it how they choose.

Reply:
I agree with @armills. It is impossible to try to work around every possible variation.

I think that updating Throttle to this behavior should also fix it without any further code:

```
if called recently:
  return None

# This is new
if call in progress:
  wait till call done
  return None

call actual function()
```

Reply:
@balloob that will not work and end in a executor deadlock.

I don't think that is a workaround. We have actual a lot of this workarounds inside code they check for `async_update`. I don't think that is nice while we should not aware of that. With this change we are clean and a platform developer can deside how that will run.